File: playwright/utils/testDataFactory.js
import { randomUUID } from "crypto";

export function uniqueEmail(prefix = "user") {
  const stamp = Date.now();
  const rand = randomUUID().slice(0, 6);
  return `${prefix}.${stamp}.${rand}@acme.test`;
}

/**
 * @param {Record<string, string>} [overrides]
 * @returns {Record<string, string>}
 */
export function buildUser(overrides = {}) {
  const stamp = String(Date.now()).slice(-6);
  return {
    firstName: "E2E",
    lastName: `User${stamp}`,
    email: uniqueEmail("user"),
    phone: `555-${stamp.slice(0, 3)}-${stamp.slice(3)}`,
    role: "Engineer",
    status: "Active",
    ...overrides
  };
}

File: src/test/java/com/automation/um/utils/TestDataFactory.java
package com.automation.um.utils;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public final class TestDataFactory {
  private TestDataFactory() {}

  public static String uniqueEmail(String prefix) {
    return prefix + "." + System.currentTimeMillis() + "." + UUID.randomUUID().toString().substring(0, 6) + "@acme.test";
  }

  public static Map<String, String> buildUser() {
    return buildUser(new HashMap<>());
  }

  public static Map<String, String> buildUser(Map<String, String> overrides) {
    String stamp = String.valueOf(System.currentTimeMillis()).substring(7);
    Map<String, String> user = new HashMap<>();
    user.put("firstName", "E2E");
    user.put("lastName", "User" + stamp);
    user.put("email", uniqueEmail("user"));
    user.put("phone", "555-" + stamp.substring(0, 3) + "-" + stamp.substring(3));
    user.put("role", "Engineer");
    user.put("status", "Active");
    user.putAll(overrides);
    return user;
  }
}
